contracts/governance/AxelarGovernance.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+
+import { AxelarExecutable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol';
+import { TimeLock } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/TimeLock.sol';
+import { IAxelarGovernance } from '../interfaces/IAxelarGovernance.sol';
+
+contract AxelarGovernance is AxelarExecutable, TimeLock, IAxelarGovernance {
+    enum Command {
+        ScheduleProposal,
+        CancelProposal
+    }
+
+    bytes32 public immutable governanceChainHash;
+    bytes32 public immutable governanceAddressHash;
+
+    // solhint-disable-next-line no-empty-blocks
+    constructor(
+        address gatewayAddress,
+        string memory governanceChain_,
+        string memory governanceAddress_,
+        uint256 minimumTimeDelay
+    ) AxelarExecutable(gatewayAddress) TimeLock(minimumTimeDelay) {
+        governanceChainHash = keccak256(bytes(governanceChain_));
+        governanceAddressHash = keccak256(bytes(governanceAddress_));
+    }
+
+    function executeProposal(address target, bytes calldata callData) external payable virtual {
+        _executeProposal(target, callData);
+    }
+
+    function _executeProposal(address target, bytes calldata callData) internal {
+        bytes32 proposalHash = keccak256(abi.encodePacked(target, callData));
+
+        _executeTimeLock(proposalHash);
+
+        (bool success, ) = target.call{ value: msg.value }(callData);
```

Comment:
hmm, msg.value should be fixed in the proposal hash when it's scheduled. Otherwise, anyone can frontrun with a different value, e.g. if we are trying to withdraw some tokens etc.